docs/docs/icicle/primitives/fri.md

```diff
@@ -0,0 +1,250 @@
+
+### Prover
+
+#### Commit and Fold Phase
+* The prover commits to the polynomial evaluations by constructing a Merkle tree.
+* A folding step is performed iteratively to reduce the polynomial degree.
+* In each step, the polynomial is rewritten using random coefficients derived from Fiat-Shamir hashing, and a new Merkle tree is built for the reduced polynomial.
+* This process continues recursively until the polynomial reaches a minimal length.
+* Currently, only a folding factor of 2 is supported.
+
+#### Proof of Work Phase (Optional)
+* If enabled, the prover is required to find a nonce such that, when hashed with the final Merkle tree root, the result meets a certain number of leading zero bits.
+
+#### Query Phase
+* Using the Fiat-Shamir transform, the prover determines the random query indices based on the previously committed Merkle roots.
+* For each sampled index, the prover provides the corresponding Merkle proof, showing that the value is part of the committed Merkle tree.
+* The prover returns all required data as the FriProof, which is then verified by the verifier.
+
+### Verifier
+* The verifier checks the Merkle proofs to ensure the sampled values were indeed committed to in the commit phase.
+* The verifier reconstructs the Fiat-Shamir challenges from the prover's commitments and verifies that the prover followed the protocol honestly.
+* The folding relation is checked for each sampled query.
+* If all checks pass, the proof is accepted as valid.
+
+
+## C++ API
+
+### Configuration structs
+There are two key configuration structs related to the Fri protocol.
+
+#### FriConfig
+The `FriConfig` struct is used to specify parameters for the FRI protocol. It contains the following fields:
+- **`stream: icicleStreamHandle`**: The CUDA stream for asynchronous execution. If `nullptr`, the default stream is used.
+- **`folding_factor: size_t`**: The factor by which the codeword is folded in each round.
+- **`stopping_degree: size_t`**: The minimal polynomial degree at which folding stops.
+- **`pow_bits: size_t`**: Number of leading zeros required for proof-of-work. If set, the optional proof-of-work phase is executed.
+- **`nof_queries: size_t`**: Number of queries computed for each folded layer of FRI.
+- **`are_inputs_on_device: bool`**: If true, the input polynomials are stored on the device (e.g., GPU); otherwise, they remain on the host (e.g., CPU).
+- **`is_async: bool`**: If true, it runs the hash asynchronously.
+- **`ext: ConfigExtension*`**: Backend-specific extensions.
+
+The default values are:
+```cpp
+// icicle/fri/fri_config.h
+struct FriConfig {
+  icicleStreamHandle stream = nullptr;
+  size_t folding_factor = 2;
+  size_t stopping_degree = 0;
+  size_t pow_bits = 16;
+  size_t nof_queries = 100;
+  bool are_inputs_on_device = false;
+  bool is_async = false;
+  ConfigExtension* ext = nullptr;
+};
+```
+> **_NOTE:_** Currently, only a folding factor of 2 is supported.
```

Comment:
Please use docusaurus admonitions for all NOTE, WARNING, INFO, etc:
https://docusaurus.io/docs/markdown-features/admonitions

```suggestion
:::note

Currently, only a folding factor of 2 is supported.

:::
```

Otherwise, its very difficult to see these in the rendered docs:

![image](https://github.com/user-attachments/assets/2bb8d63f-508e-4524-bb55-a91845d8d83d)